explorer/events.go

```diff
@@ -65,6 +68,56 @@ func (EventV1ContractResolution) isEvent() bool { return true }
 func (EventV2Transaction) isEvent() bool        { return true }
 func (EventV2ContractResolution) isEvent() bool { return true }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (e *Event) UnmarshalJSON(b []byte) error {
+	var je struct {
```

Comment:
I think you missed `confirmations`, maybe we should consider adding a unit test that marshals/unmarshals an event since it's quite an odd type